### PR TITLE
fix react errors and unit test

### PR DESCRIPTION
**Pull request by mistermendez**

### CODE REVIEW

File: js/components/colorSelector.js [hunk 1]
```diff
@@ -11,20 +10,24 @@ var ColorSelector = React.createClass({
...
+
   render: function() {
     var colorItems = [];
     for (var i = 0; i < this.props.colors.length; i++) {
       colorItems.push(
-        <div className={this.setClassname(this.props.colors[i])}>
-          <div className={"oo-color-item oo-color-item-" + this.props.colors[i]} onClick={this.props.onColorChange.bind(this, this.props.colors[i])} key={i}></div>
+        <div className={this.setClassname(this.props.colors[i])} key={i}>
+          <a className={"oo-color-item oo-color-item-" + this.props.colors[i]} onClick={this.handleColorSelection.bind(this, this.props.colors[i])}></a>
         </div>
```
Comment (mistermendez) [inline js/components/colorSelector.js]: * resolve bind error from react
  * onClick event handlers need to be on clickable elements (ie - link, button) to work on mobile browsers

File: js/components/colorSelector.js [hunk 2]
```diff
@@ -11,20 +10,24 @@ var ColorSelector = React.createClass({
...
     }
 
-    return(
-      <span className="oo-color-selector">
+    return (
+      <div className="oo-color-selector">
         {colorItems}
-      </span>
+      </div>
     );
```
Comment (mistermendez) [inline js/components/colorSelector.js]: * block elements (div) cannot be contained by inline elements (span)

File: js/components/dataSelector.js [hunk 3]
```diff
@@ -59,7 +62,7 @@ var DataSelector = React.createClass({
       var dataContentBlocks = [];
       for (var i = 0; i < dataItems.length; i++) {
         dataContentBlocks.push(
-          <a className={this.setClassname(dataItems[i])} onClick={this.props.onDataChange.bind(this, dataItems[i])} key={i}>
+          <a className={this.setClassname(dataItems[i])} onClick={this.handelDataSelection.bind(this, dataItems[i])} key={i}>
             <span className="oo-data">{dataItems[i]}</span>
```
Comment (mistermendez) [inline js/components/dataSelector.js]: * resolve react bind error
  * resolve react key error

**Comment by rchalooyala** [comment]
thanks

**Pull request merged**